### PR TITLE
Auth based on google token

### PR DESCRIPTION
**Pull request by LBHMGeorgieva**
1. Implemented second authorization workflow based on 'hackneyToken' jwt generated by the Google auth solution. 
2. Amended existing flow to assume role for retrieving API name from API Gateway in order to support cross-account
3. Added DynamoDB implementation as this is where authentication data mapping will be saved

Still to be done:

1. Implement better exception handling

### CODE REVIEW

File: ApiAuthVerifyToken/V1/Gateways/AwsApiGateway.cs [hunk 1]
```diff
@@ -10,15 +12,24 @@ public class AwsApiGateway : IAwsApiGateway
...
+            try
             {
-                var response = _client.GetRestApiAsync(new GetRestApiRequest { RestApiId = apiId }).Result;
-
-                if (response == null)
+                using (_client = new AmazonAPIGatewayClient(awsCredentials))
                 {
-                    throw new AwsApiNotFoundException();
+                    LambdaLogger.Log($"Begin getting API name for API id {apiId}");
+                    var response = _client.GetRestApiAsync(new GetRestApiRequest { RestApiId = apiId }).Result;
```
Comment (LBHBomotoso) [inline ApiAuthVerifyToken/V1/Gateways/AwsApiGateway.cs:20]: Don't you want to make this async?

File: ApiAuthVerifyToken/V1/Domain/APIDataUserFlow.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace ApiAuthVerifyToken.V1.Domain
+{
+    public class APIDataUserFlow
+    {
+        public Guid Id { get; set; }
+        public string ApiName { get; set; }
+        public string Environemnt { get; set; }
```
Comment (LBHMKeyworth) [inline ApiAuthVerifyToken/V1/Domain/APIDataUserFlow.cs:11]: Typo - `Environment`

File: ApiAuthVerifyToken/V1/Infrastructure/APIDataUserFlowDbEntity.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace ApiAuthVerifyToken.V1.Infrastructure
+{
+    [DynamoDBTable("APIAuthenticatorData", LowerCamelCaseProperties = true)]
+    public class APIDataUserFlowDbEntity
+    {
+        [DynamoDBHashKey]
+        public string ApiName { get; set; }
+        [DynamoDBRangeKey("environment")]
+        public string Environemnt { get; set; }
```
Comment (LBHMKeyworth) [inline ApiAuthVerifyToken/V1/Infrastructure/APIDataUserFlowDbEntity.cs:14]: Here too `Environment`

**Review by LBHBomotoso — APPROVED** [review]

File: ApiAuthVerifyToken.Tests/V1/AcceptanceTests/VerifyTokenAcceptanceTests.cs [hunk 4]
```diff
@@ -71,16 +86,72 @@ public void FunctionShouldReturnAPIGatewayCustomAuthorizerPolicyWithDenyEffectWh
...
             Environment.SetEnvironmentVariable("jwtSecret", _fixture.Create<string>());
             var result = _classUnderTest.VerifyToken(lambdaRequest);
 
             result.Should().BeOfType<APIGatewayCustomAuthorizerResponse>();
             result.PolicyDocument.Statement.First().Effect.Should().Be("Deny");
         }
+        #region UserFlow
+        [Test]
+        public void FunctionShouldReturnAPIGatewayCustomAuthorizerPolicyWithAllowEffectWhenAuthenticationSuccesdsForUserFlow()
+        {
```
Comment (LBHBomotoso) [inline ApiAuthVerifyToken.Tests/V1/AcceptanceTests/VerifyTokenAcceptanceTests.cs:103]: Can this be shortened? =]

**Review by LBHMKeyworth — APPROVED** [review]
Looks good to me.

Great stuff!!!

**Pull request merged**